src/lib/mina_lib/mina_lib.ml

```diff
@@ -1495,6 +1495,126 @@ let start_filtered_log ~commit_id
       () ;
     Ok () )
 
+let fetch_completed_snarks (module Context : CONTEXT) snark_pool network
+    top_block get_most_recent_valid_block =
+  let open Context in
+  let open Network_peer in
+  let%bind all_peers = Mina_networking.peers network in
+  Deferred.List.iter
```

Comment:
Should we be using fold_until here and exit when all the requested work is fetched?
https://github.com/MinaProtocol/mina/blob/86135da5c95a07690218aabdae19d318c3bae27c/src/lib/mina_stdlib/deferred.ml#L14

If the goal is to fetch work from different peers to maximize getting relevant+cheap snarks then we should probably cap the # of peers. Some nodes can have large number of peers. For example, we set seeds to have 200 max connections (but it can go higher that that at any point before it adjusts to the limit)

Reply:
ok sure i can do that, i will keep it to 5 for now.